nntrainer/layers/permute_layer.h

```diff
@@ -14,11 +14,34 @@
 #define __PERMUTE_LAYER_H__
 
 #include <array>
+#include <string>
 
 #include <base_properties.h>
 #include <layer_internal.h>
 #include <node_exporter.h>
 namespace nntrainer {
+
+namespace props {
```

Comment:
Are we going to have this props namespace? if it is we need to think more. More code is coming.. so.. maybe we need more namespace like layer.. and optimizer.

Reply:
props namespace are used to explicitly tell it is derived from a property class and can be dealt as other properties are used.
This namespace is used inside `common_properties.h` `base_properties.h` and some custom properties.

```cpp
   props::Name name; // we can explicitly know from the code it is property, and we can expect some common functions like to_string, from_string with this class.
```

vs 
```cpp
  Name name; // we don't know it's a class or a property-ish class.
```
I think it's a policy we can decide. I wonder if we should not use namespace for the props?

Reply:
No. I'm saying, If we need, we have to use it other places to maintain consistency.

Reply:
`common_properties.h` has props shared across layers.
and each layer can have its own properties. Should we not use namespace for that?